File: TravelPlanner.ServiceFabric/TravelPlanner.TravelService/Models/Activity.cs
namespace TravelPlanner.TravelService.Models;

public class Activity
{
    public Guid Id { get; set; }
    public Guid PlanId { get; set; }
    public Guid? DestinationId { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public string? Time { get; set; }
    public string Location { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal EstimatedCost { get; set; }
    public string Status { get; set; } = "planned"; // planned, reserved, completed, cancelled

    public TravelPlan Plan { get; set; } = null!;
    public Destination? Destination { get; set; }
}

File: travel-planner-frontend/src/models/Activity.js
export class Activity {
  constructor(id, planId, destinationId, name, date, time, location, description, estimatedCost, status) {
    this.id = id;
    this.planId = planId;
    this.destinationId = destinationId;
    this.name = name;
    this.date = date;
    this.time = time;
    this.location = location;
    this.description = description;
    this.estimatedCost = estimatedCost;
    this.status = status;
  }
}